From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KurumiFake <kurumiisshidowife@gmail.com>
Date: Wed, 1 Jun 2022 08:58:32 +0700
Subject: [PATCH] strip version command even moar


diff --git a/src/main/java/org/bukkit/command/defaults/VersionCommand.java b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
index e0596ea71c517199bacfc122d23b3a195e3d6ba8..68677380a1f247fc84f8ff68505a5b5309d2ccfc 100644
--- a/src/main/java/org/bukkit/command/defaults/VersionCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
@@ -245,11 +245,11 @@ public class VersionCommand extends BukkitCommand {
             Component.newline(),
             msg
         );
-        this.versionMessage = Component.text()
+        /*this.versionMessage = Component.text()
             .append(message)
             .hoverEvent(Component.text("Click to copy to clipboard", net.kyori.adventure.text.format.NamedTextColor.WHITE))
             .clickEvent(net.kyori.adventure.text.event.ClickEvent.copyToClipboard(net.kyori.adventure.text.serializer.plain.PlainComponentSerializer.plain().serialize(message)))
-            .build();
+            .build();*/
         // Paper end
         versionLock.lock();
         try {
@@ -264,7 +264,7 @@ public class VersionCommand extends BukkitCommand {
         }
     }
 
-    private static int getDistance(@NotNull String repo, @NotNull String hash) {
+    /*private static int getDistance(@NotNull String repo, @NotNull String hash) {
         try {
             BufferedReader reader = Resources.asCharSource(
                     new URL("https://hub.spigotmc.org/stash/rest/api/1.0/projects/SPIGOT/repos/" + repo + "/commits?since=" + URLEncoder.encode(hash, "UTF-8") + "&withCounts=true"),
@@ -283,5 +283,5 @@ public class VersionCommand extends BukkitCommand {
             e.printStackTrace();
             return -1;
         }
-    }
+    }*/
 }
