From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KurumiFake <kurumiisshidowife@gmail.com>
Date: Thu, 17 Mar 2022 06:53:00 +0000
Subject: [PATCH] chaos


diff --git a/src/main/java/me/noahvdaa/dripstone/SetMaxTPSCommand.java b/src/main/java/me/noahvdaa/dripstone/SetMaxTPSCommand.java
index d31951e83ff3256420c57eac8f9fc16183f1565d..1a8aaa5f16b3019d1076fbf8df8aa1467ac43ee2 100644
--- a/src/main/java/me/noahvdaa/dripstone/SetMaxTPSCommand.java
+++ b/src/main/java/me/noahvdaa/dripstone/SetMaxTPSCommand.java
@@ -26,7 +26,7 @@ public class SetMaxTPSCommand extends Command {
     public boolean execute(CommandSender sender, String commandLabel, String[] args) {
         if (!testPermission(sender)) return true;
 
-        int targetMSPT = 50;
+        float targetMSPT = 50;
         int resetTPSIn = -1;
 
         if (args.length == 0) {
@@ -37,13 +37,13 @@ public class SetMaxTPSCommand extends Command {
                 return true;
             }
 
-            int targetTPS = Integer.parseInt(args[0]);
+            float targetTPS = Float.parseFloat(args[0]); // Lazinity - :troll:
 
-            // Don't allow negative or zero TPS.
+            /* // Don't allow negative or zero TPS. // Lazinity - nope lemme break my server :troll:
             if (targetTPS < 1) {
                 sender.sendMessage("§cThe target TPS must be at least 1.");
                 return true;
-            }
+            }*/
 
             targetMSPT = 1000 / targetTPS;
 
