From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Etil <81570777+etil2jz@users.noreply.github.com>
Date: Mon, 3 Jan 2022 16:03:27 +0100
Subject: [PATCH] (Mirai) Configurable server metrics


diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
index 7a86357ea75db7263598b79d31d4ae9abe5e4b50..1e2683e45d17b1f3b35cf7293f413e974eb1022f 100644
--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
@@ -96,7 +96,7 @@ public class PaperConfig {
             MinecraftServer.getServer().server.getCommandMap().register(entry.getKey(), "Paper", entry.getValue());
         }
 
-        if (!metricsStarted) {
+        if (!metricsStarted && me.kurumifake.lazinity.LazinityConfig.serverMetrics) { // Mirai - configurable server metrics
             Metrics.PaperMetrics.startMetrics();
             metricsStarted = true;
         }
diff --git a/src/main/java/me/kurumifake/lazinity/LazinityConfig.java b/src/main/java/me/kurumifake/lazinity/LazinityConfig.java
index fedac8069ec87c3e9bc7e5a0a7381e362734bb5b..3d58dd42f02670ea24d90417e6e75cd78dc86eab 100644
--- a/src/main/java/me/kurumifake/lazinity/LazinityConfig.java
+++ b/src/main/java/me/kurumifake/lazinity/LazinityConfig.java
@@ -160,4 +160,9 @@ public final class LazinityConfig {
         checkVehicleMovedQuickly = getBoolean("movement-checks.vehicle-moved-quickly", checkVehicleMovedQuickly);
         checkVehicleMovedWrongly = getBoolean("movement-checks.vehicle-moved-wrongly", checkVehicleMovedWrongly);
     }
+
+    public static boolean serverMetrics = false;
+    private static void allowServerMetrics() {
+        serverMetrics = getBoolean("bstats-metrics", serverMetrics);
+    }
 }
